cli/src/main/java/de/jplag/cli/CLI.java

```diff
@@ -83,153 +56,164 @@ public final class CLI {
-        if (!(factory instanceof CollectedLoggerFactory collectedLoggerFactory))
-            return;
-        collectedLoggerFactory.finalizeInstances();
-    }
-
     /**
-     * Creates the command line interface and initializes the argument parser.
+     * Creates a new instance
      */
     public CLI() {
-        parser = ArgumentParsers.newFor(PROGRAM_NAME).build().defaultHelp(true).description(generateDescription());
-        CliGroupHelper groupHelper = new CliGroupHelper(parser);
-        for (CommandLineArgument argument : CommandLineArgument.values()) {
-            argument.parseWith(parser, groupHelper);
-        }
+        this.options = new CliOptions();
+        this.commandLine = new CommandLine(options);
+
+        buildSubcommands().forEach(commandLine::addSubcommand);
+
+        this.commandLine.getHelpSectionMap().put(SECTION_KEY_FOOTER, help -> generateDescription());
+        this.commandLine.setAllowSubcommandsAsOptionParameters(true);
+    }
+
+    private List<CommandSpec> buildSubcommands() {
+        return LanguageLoader.getAllAvailableLanguages().values().stream().map(language -> {
+            CommandSpec command = CommandSpec.create().name(language.getIdentifier());
+
+            for (LanguageOption<?> option : language.getOptions().getOptions()) {
+                command.addOption(OptionSpec.builder("--" + option.getName()).type(option.getType().getJavaType())
+                        .description(option.getDescription()).build());
+            }
+            command.addOption(OptionSpec.builder("-h", "--help").usageHelp(true).build());
+            command.addPositional(
+                    CommandLine.Model.PositionalParamSpec.builder().type(List.class).auxiliaryTypes(File.class).hidden(true).required(false).build());
+
+            return command;
+        }).toList();
     }
 
     /**
-     * Parses an array of argument strings.
-     * @param arguments is the array to parse.
-     * @return the parsed arguments in a {@link Namespace} format.
+     * Parses the options from the given command line arguments. Also prints help pages when requested.
+     * @param args The command line arguments
+     * @return the parse result generated by picocli
      */
-    public Namespace parseArguments(String[] arguments) {
+    public ParseResult parseOptions(String... args) {
         try {
-            return parser.parseArgs(arguments);
-        } catch (ArgumentParserException exception) {
-            parser.handleError(exception);
+            ParseResult result = commandLine.parseArgs(args);
+            if (result.isUsageHelpRequested() || (result.subcommand() != null && result.subcommand().isUsageHelpRequested())) {
+                commandLine.getExecutionStrategy().execute(result);
+            }
+            return result;
+        } catch (CommandLine.PicocliException e) {
+            logger.error("Error during parsing", e);
             System.exit(1);
```

Comment:
Ah, well, always interesting to see what we previously missed in PR reviews! It would be great if you could improve this/check if ExitExceptions are viable.